app/views/topology/service.js

```diff
@@ -1220,6 +1220,7 @@ YUI.add('juju-topology-service', function(Y) {
                    .size([width, height])
                    .value(function(d) { return Math.max(d.unit_count, 1); })
                    .padding(300)
```

Comment:
Could you add some comments here as to why each one of these chainings is required? 

Also re-your test, This return a data structure that you can compare against by chance?